azkaban-common/src/main/java/azkaban/imagemgmt/daos/ImageVersionDaoImpl.java

```diff
@@ -354,8 +354,27 @@ public void updateImageVersion(final ImageVersion imageVersionRequest)
       queryBuilder.append(" where iv.id = ? and iv.type_id = it.id and it.name = ?");
       params.add(imageVersionRequest.getId());
       params.add(imageVersionRequest.getName());
-      final int updateCount = this.databaseOperator.update(queryBuilder.toString(),
-          Iterables.toArray(params, Object.class));
+      int updateCount = 0;
+      if (imageVersionRequest.getState() != State.ACTIVE) {
```

Comment:
Shall we also put a check for state machine that from which all state transition is possible to ACTIVE?

Reply:
That is out of scope for this change, let me create a ticket to track it.